invenio_records_resources/services/files/schema.py

```diff
@@ -85,7 +85,18 @@ def fields_from_file_obj(self, data, **kwargs):
         if data.file:
             # mandatory fields
             data["storage_class"] = data.file.storage_class
-            data["uri"] = data.file.uri  # TODO: only serialize for external?
+            data["uri"] = data.file.uri
+
+            # If Local -> remove uri as it contains internal file storage info
+            if data["storage_class"] == TransferType.LOCAL:
```

Comment:
Could we use the `Transfer.` class factory with the `is_completed` function? That way we avoid exposing/coupling the TransferType categories.